config/routes.rb

```diff
@@ -18,14 +18,11 @@
     resources :subscriptions, only: [:index, :destroy]
   end
 
-  resources :customers, only: [:create]
-  resources :plans, only: [:index], constraints: SubscriptionsUserConstraint.new
-  resources :products, only: [:index, :show]
-  resources :subscriptions, only: [:create]
-
-  post '/subscriptions/finalize' => 'subscriptions#finalize'
+  get '/' => 'subscribe#index'
+  get '.json' => 'subscribe#index'
```

Comment:
I don't think this is necessary, the previous route declaration is not limited to a format, so it should respond on `.json` too.

Reply:
Somehow it turned out to be. It works fine in production but tests fail without it. Maybe it's something with how the engine is set up? I'd have to take a look.